server/utils/backup/backup.utils.ts

```diff
@@ -72,27 +72,100 @@ export interface VerifyRecoveryFileSignature extends RecoveryFileData {
   recoveryFileServerSignature: string;
 }
 
-async function verifyRecoveryFileSignature({
-  recoveryFileServerSignature,
-  ...recoveryFileData
-}: VerifyRecoveryFileSignature) {
-
-  const publicKey = await crypto.subtle.importKey(
-    "spki",
-    Buffer.from(Config.BACKUP_FILE_PUBLIC_KEY, "base64"),
-    IMPORT_KEY_ALGORITHM,
-    true,
-    ["sign"],
-  );
-
-  const recoveryFileRawData = getRecoveryFileSignatureRawData(recoveryFileData);
-  const recoveryFileRawDataBuffer = Buffer.from(recoveryFileRawData);
+export async function verifyRecoveryFileSignature({
+  walletId,
+  recoveryBackupShareHash,
+  recoveryFileServerSignature
+}: {
+  walletId: string;
+  recoveryBackupShareHash: string;
+  recoveryFileServerSignature: string;
+}): Promise<boolean> {
+  try {
+    console.log("Verifying recovery file signature for wallet:", walletId);
+    
+    // Handle both formats of recoveryFileServerSignature
+    const signature = recoveryFileServerSignature.startsWith('v1.') 
+      ? recoveryFileServerSignature.substring(3) 
+      : recoveryFileServerSignature;
+    
+    // Create a signature object
+    const signatureBuffer = Buffer.from(signature, 'base64');
+    
+    // Create the verification data using the same function used when generating the signature
+    const verificationData = getRecoveryFileSignatureRawData({
+      walletId,
+      recoveryBackupShareHash
+    });
+    
+    try {
+      // Try RSA-PSS first (newer format)
```

Comment:
Why did this change? In any case, if this change is needed, the old format can be removed because all this is pre-launch, so no one will be using the old one.

Reply:
I was testing out a newer format based on some research I made.  I reverted back to the old format as I don't think we need to switch to that format.